src/hooks/shapediver/viewer/useOutputDrawingTools.ts

```diff
@@ -0,0 +1,99 @@
+import { IGeometryData, IOutputApi, ISessionApi, ITreeNode, IViewportApi } from "@shapediver/viewer";
+import { useCallback, useEffect, useRef } from "react";
+import { IDrawingToolsApi, PlaneRestrictionProperties, RESTRICTION_TYPE, createDrawingTools } from "@shapediver/viewer.features.drawing-tools";
+import { useShapeDiverStoreViewer } from "store/useShapeDiverStoreViewer";
+import { CustomizationPropertiesOptional } from "@shapediver/viewer.features.drawing-tools/dist/business/implementation/DrawingToolsManager";
+import { useOutputNode } from "./useOutputNode";
+
+/**
+ * Hook allowing to create the drawing tools.
+ * 
+ * Makes use of {@link useOutputNode}.
+ * 
+ * @param sessionId 
+ * @param viewportId 
+ * @param outputIdOrName 
+ */
+export function useOutputDrawingTools(sessionId: string, viewportId: string, outputIdOrName: string) : {
+	/**
+	 * API of the output
+	 * @see https://viewer.shapediver.com/v3/latest/api/interfaces/IOutputApi.html
+	 */
+	outputApi: IOutputApi | undefined,
+
+	drawingToolsApiRef: React.MutableRefObject<IDrawingToolsApi | null>
+} {
+
+	/**
+	 * Create the callbacks for the drawing tools
+	 * @ALEX I'm not sure how to actually get parameters, if not via the API
+	 */
+	const onFinish = async (geometryData: IGeometryData) => {
+		if(!sessionApiRef.current) return;
+		console.log("onFinish"); 
+        
+		const positionArray = geometryData.primitive.attributes["POSITION"].array;
+
+		// get all points
+		const points = [];
+		for (let i = 0; i < positionArray.length; i += 3) {
+			points.push([positionArray[i], positionArray[i + 1], positionArray[i + 2]]);
+		}
+
+		sessionApiRef.current.getParameterByName("points")[0].value = JSON.stringify({ points: points });
+		await sessionApiRef.current.customize();
+	};
+	const onCancel = () => { console.log("onCancel"); };
+
+
+	// callback which will be executed on update of the output node
+	// @ALEX: it seems like sometimes this update is not triggered when the node changed, I probably have to investigate this further
+	const callback = useCallback( (newNode?: ITreeNode) => {
+		if(!newNode) return;
+		if(drawingToolsApiRef.current) drawingToolsApiRef.current.close();
+		customizationPropertiesRef.current = outputApiRef.current?.content![0].data as CustomizationPropertiesOptional;
+
+		if(viewportApiRef.current && customizationPropertiesRef.current && outputApiRef.current) {
```

Comment:
@MajorMeerkatThe3rd instead of using useRef, pass the viewportApi as a dependency of the useCallback hook